lib/registrant.rb

```diff
@@ -0,0 +1,28 @@
+class Registrant
+  attr_reader :name, :age, :permit, :earn_permit 
+  
+  attr_accessor :license_data
+  
+  def initialize(name, age, permit = false)
+    @name = name
+    @age = age
+    @permit = permit
+    @license_data = {
+      :written=>false, 
+      :license=>false, 
+      :renewed=>false
+    }
+  end
+
+  def permit?
+    if @permit == true || @earn_permit == true
```

Comment:
I'm not sure this line of code is working the way you intended. The ivar `@earn_permit` will always return `nil` because it isn't defined anywhere. 
Digging a little deeper on this logic, we don't need to consider the `earn_permit` method in this conditional statement at all. This is because once the `earn_permit` method is called, it will update the `@permit` attribute, which is what's being checked here anyway.